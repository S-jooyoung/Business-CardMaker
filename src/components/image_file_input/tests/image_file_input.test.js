// import dependencies
import React from "react";
import userEvent from "@testing-library/user-event";

// import custom-component
import ImageFileInput from "../image_file_input";

// import snapshot-testing methods
import { render, screen } from "@testing-library/react";

describe("Image_file_input", () => {
  let ImageFileInputComponent;
  let name = "test";
  let upload;
  let onFileChange;

  beforeEach(() => {
    upload = jest.fn();
    onFileChange = jest.fn();
    ImageFileInputComponent = (
      <ImageFileInput upload={upload} onFileChange={onFileChange} name={name} />
    );
  });

  describe("Button Click", () => {
    const file = new File(["hello"], "hello.png", { type: "image/png" });
    let input;
    let button;
    beforeEach(() => {
      render(ImageFileInputComponent);
      input = screen.getByTestId("test-onChange");
      button = screen.getByTestId("test-button");
    });

    it("upload file", () => {
      userEvent.upload(input, file);

      expect(input.files[0]).toStrictEqual(file);
      expect(input.files.item(0)).toStrictEqual(file);
      expect(input.files).toHaveLength(1);
    });

    it("enables filename when file are entered", () => {
      expect(button.innerHTML).toBe("test");
    });
  });
});
