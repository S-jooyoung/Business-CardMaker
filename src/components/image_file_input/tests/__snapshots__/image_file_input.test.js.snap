// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image_file_input redner 1`] = `
<div
  className="container"
>
  <input
    accept="image/*"
    className="input"
    data-testid="test-onChange"
    name="file"
    onChange={[Function]}
    type="file"
  />
  <button
    className="button yellow"
    data-testid="test-button"
    onClick={[Function]}
  >
    test
  </button>
</div>
`;
